### PR TITLE
Add initial bindings for Log TableView

### PR DESCRIPTION
**Pull request by willsb**
This binds the UITableView to the Observable collection. The cells themselves are not bound to anything yet. This will come in a different PR, since some Model conversion will be needed.

Feel free to squash once approved.

### CODE REVIEW

File: Toggl.Daneel/Views/TimeEntryViewCell.cs [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 
 namespace Toggl.Daneel.Views
 {
-    public partial class TimeEntryViewCell : UITableViewCell
+    public partial class TimeEntryViewCell : MvxTableViewCell
     {
-        public static readonly NSString Key = new NSString(nameof(TimeEntryViewCell));
-        public static readonly UINib Nib;
+        private static readonly NSString Key = new NSString(nameof(TimeEntryViewCell));
+        private static readonly UINib Nib;
```
Comment (arturs105) [inline Toggl.Daneel/Views/TimeEntryViewCell.cs]: Why are these made private? I assume we won't be registering this cell for reuse for the `UITableView`, but instead use some bindings?

File: Toggl.Daneel/Views/TimeEntryViewCell.cs [hunk 2]
```diff
@@ -1,13 +1,20 @@
 ﻿using System;
+using System.Linq;
 using Foundation;
+using MvvmCross.Binding.BindingContext;
```
Comment (arturs105) [inline Toggl.Daneel/Views/TimeEntryViewCell.cs]: Unnecessary `using` statement

File: Toggl.Daneel/Views/TimeEntryViewCell.cs [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+using MvvmCross.Binding.iOS.Views;
+using Toggl.Multivac.Models;
```
Comment (arturs105) [inline Toggl.Daneel/Views/TimeEntryViewCell.cs]: Unnecessary

File: Toggl.Daneel/Views/TimeEntryLogHeaderViewCell.cs [hunk 4]
```diff
@@ -1,13 +1,18 @@
 ﻿using System;
+using System.Linq;
 using Foundation;
+using MvvmCross.Binding.iOS.Views;
```
Comment (arturs105) [inline Toggl.Daneel/Views/TimeEntryLogHeaderViewCell.cs]: Unnecessary `using` statement

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.Linq;
+using Foundation;
+using MvvmCross.Binding.ExtensionMethods;
+using MvvmCross.Binding.iOS.Views;
+using MvvmCross.Platform.Core;
+using Toggl.Daneel.Views;
+using Toggl.Foundation.MvvmCross.Converters;
+using Toggl.Multivac.Models;
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: Multiple unnecessary using statement.

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            => GetGroupAt(section).Count();
+
+        protected virtual IGrouping<TKey, TType> GetGroupAt(nint section)
+            => GroupedItems.ElementAt((int)section);
+
+        protected override object GetItemAt(NSIndexPath indexPath)
+            => GroupedItems.ElementAt(indexPath.Section).ElementAt((int)indexPath.Item);
+
+        protected UITableViewHeaderFooterView GetOrCreateHeaderViewFor(UITableView tableView)
+            => TableView.DequeueReusableHeaderFooterView(headerCellIdentifier);
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: This method receives a `UITableView tableview`, but instead uses `TableView` property from `MvxBaseTableViewSource`. Is this intended?

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+        protected override UITableViewCell GetOrCreateCellFor(UITableView tableView, NSIndexPath indexPath, object item)
+            => TableView.DequeueReusableCell(CellIdentifier, indexPath);
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: Same as above

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+            tableView.SeparatorStyle = UITableViewCellSeparatorStyle.None;
+        }
+
+        public override UIView GetViewForHeader(UITableView tableView, nint section)
+        {
+            var grouping = GetGroupAt(section);
+
+            var cell = GetOrCreateHeaderViewFor(tableView);
+            if (cell is IMvxBindable bindable)
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: Is this a new C# feature?

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+            return cell;
+        }
+
+        public override nint NumberOfSections(UITableView tableView)
+            => ItemsSource.Count();
+
+        public override nint RowsInSection(UITableView tableview, nint section)
+            => GetGroupAt(section).Count();
+
+        protected virtual IGrouping<TKey, TType> GetGroupAt(nint section)
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: Is this class meant to be inherited from?

File: Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs [hunk 10]
```diff
@@ -24,10 +27,18 @@ public override void ViewDidLoad()
...
             base.ViewDidLoad();
 
+            //Empty state button config
             EmptyStateButton.Layer.BorderWidth = 1;
             EmptyStateButton.Layer.BorderColor = Color.TimeEntriesLog.ButtonBorder.ToNativeColor().CGColor;
             EmptyStateButton.SetTitle(Resources.TimeEntriesLogEmptyStateButton, UIControlState.Normal);
 
+            //TableView config
+            var source = new GroupBindingTableViewSource<DateTime, ITimeEntry>(
+                TimeEntries, nameof(TimeEntryLogHeaderViewCell), nameof(TimeEntryViewCell)
```
Comment (arturs105) [inline Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs]: I suggest changing the name to `TimeEntriesTableView`, this is confusing.

File: Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs [hunk 11]
```diff
@@ -37,6 +48,9 @@ public override void ViewDidLoad()
             bindingSet.Bind(EmptyStateTextLabel).To(vm => vm.EmptyStateText);
             bindingSet.Bind(EmptyStateTitleLabel).To(vm => vm.EmptyStateTitle);
 
+            //Timeline
```
Comment (arturs105) [inline Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs]: Is this really timeline or time entry log?

File: Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs [hunk 12]
```diff
@@ -8,14 +8,17 @@
 using Toggl.Foundation;
 using Toggl.Foundation.MvvmCross.Helper;
 using Toggl.Foundation.MvvmCross.ViewModels;
```
Comment (arturs105) [inline Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs]: Unnecessary `using` statements

File: Toggl.Daneel/Views/TimeEntryViewCell.cs (hunk 1 above)
Comment (willsb) [inline Toggl.Daneel/Views/TimeEntryViewCell.cs]: We will register the cell for reuse! The bindings will be done by swapping the `BindingContext` of the cell.

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs (hunk 8 above)
Comment (willsb) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: Yes it is :)

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs (hunk 9 above)
Comment (willsb) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: If needed be, one can inherit from it

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs (hunk 6 above)
Comment (willsb) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: Nope 🙈

File: Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs (hunk 12 above)
Comment (willsb) [inline Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs]: I beg to differ: ViewModels is used by `TimeEntriesLogViewModel` and Helper is used by `Color.TimeEntriesLog`

File: Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs (hunk 11 above)
Comment (willsb) [inline Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs]: Naming  = Hard

File: Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs (hunk 10 above)
Comment (willsb) [inline Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs]: 👍

File: Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs (hunk 8 above)
Comment (amulware) [inline Toggl.Daneel/ViewSources/GroupBindingTableViewSource.cs]: 💅

File: Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs (hunk 12 above)
Comment (arturs105) [inline Toggl.Daneel/ViewControllers/TimeEntriesLogViewController.cs]: Sorry, VS got confused.

**Pull request merged**